File: frontend/lib/models/tent_shape.dart
class TentShape {
  final String id;
  final String name;
  final int displayOrder;
  final bool isActive;

  const TentShape({
    required this.id,
    required this.name,
    required this.displayOrder,
    required this.isActive,
  });

  factory TentShape.fromJson(Map<String, dynamic> json) {
    return TentShape(
      id: json['id'] as String,
      name: json['name'] as String,
      displayOrder: json['displayOrder'] as int,
      isActive: json['isActive'] as bool? ?? true,
    );
  }
}

File: backend/ScoutBoxApi/Models/Entities/TentShape.cs
namespace ScoutBoxApi.Models.Entities;

public class TentShape : IHasName
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsActive { get; set; }
    public int DisplayOrder { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<TentShapePart> TentShapeParts { get; set; } = new List<TentShapePart>();
    public ICollection<Tent> Tents { get; set; } = new List<Tent>();
}
